throw an appropriate error when payload validation is used with GET or HEAD routes

damirv opened the pull request:
When a user mistakenly uses a payload validation on a GET or HEAD route, an appropriate error is given, instead of

```
TypeError: Cannot read property 'parse' of null
   at new module.exports.internals.Route (... node_modules/hapi/lib/route.js:126:73)
```

This might help new users that make this mistake more quickly to uncover what's going on.

Closes #2347

hueniverse commented:
It needs to ignore it at the route lifecycle setup step, not assert. This is because you can now set a global route validation default for all routes.

damirv commented:
Sure, where can I find the route lifecycle setup code?

hueniverse commented:
Setup is here: https://github.com/hapijs/hapi/blob/master/lib/route.js#L247
But method can be `*` so need to skip it here: https://github.com/hapijs/hapi/blob/master/lib/validation.js#L19

damirv commented:
I will try to fix this tomorrow.

damirv commented:
I am unfamiliar with global route validation, is there any documentation on that? (couldn't find anything)

Currently, I have added it to the two suggested places. However, I also added the check where I first made the change, because that is where error originally occurred.

I also included a simple test case. If you point me to the doc on global route validation, I can add that test case as well and pass the 100% coverage requirement.

hueniverse commented:
Solved slightly differently.
